app.js

```diff
@@ -724,10 +724,27 @@ function publishDiscovery( device, index ){
 					}
 */
 					var discoveryTopic = `${CONFIG.discovery_prefix}${parameter.component}/ener314rt/${object_id}/config`;
-//					var dmsg = Object.assign({ uniq_id: `${unique_id}`, "~": `${CONFIG.topic_stub}`, name: `${name}`, mf: 'energenie', sw: 'mqtt-ener314rt' },
-					var dmsg = Object.assign( { device: { name: `${device_name}`, ids: [`ener314rt-${device.deviceId}`], mdl: `${device_defaults.mdl}`, mf: 'Energenie', sw: `mqtt-ener314rt ${APP_VERSION}` }, 
-											uniq_id: `${unique_id}`, "~": `${CONFIG.topic_stub}${device.productId}/${device.deviceId}/`, name: `${entity_name}`, availability_topic: `${CONFIG.topic_stub}availability/state` },
-											parameter.config,);
+
+					var dmsg = Object.assign({
+						device: {
+							name: `${device_name}`,
+							ids: [`ener314rt-${device.deviceId}`],
+							mdl: `${device_defaults.mdl}`,
+							mf: `Energenie`,
+							sw: `${APP_VERSION}`,
+							hw: `mqtt-ener314rt`
+						},
+						uniq_id: `${unique_id}`,
+						"~": `${CONFIG.topic_stub}${device.productId}/${device.deviceId}/`,
+						name: `${entity_name}`,
+						availability_topic: `${CONFIG.topic_stub}availability/state`,
+						origin: {
```

Comment:
Not seen `origin` before.  Where is the documentation for this?

Reply:
https://www.home-assistant.io/integrations/mqtt/
![image](https://github.com/Achronite/mqtt-energenie-ener314rt/assets/5207490/e41b2653-5289-46e4-ab52-b69eae9cc63b)
![image](https://github.com/Achronite/mqtt-energenie-ener314rt/assets/5207490/89079bcc-9c5b-48ca-83da-f8cc2b9006cf)

Reply:
OK, pls use the abbreviations though, `o` etc, it cuts down on network traffic and MQTT usage.

Reply:
Done